src/parquet/column/scanner.h

```diff
@@ -78,7 +78,7 @@ class Scanner {
   int level_offset_;
   int levels_buffered_;
 
-  std::vector<uint8_t> value_buffer_;
+  OwnedMutableBuffer value_buffer_;
```

Comment:
I figure most of these will have to be revisited as we thread a top-level MemoryPool through each of these classes (any allocated memory would be requested from the active MemoryPool in that context rather than the default one)